apps/site/pages/en/learn/modules/package-configuration.mdx

```diff
@@ -0,0 +1,597 @@
+
+<table>
+  <thead>
+    <tr>
+      <th>You as a package author write</th>
+      <th>Consumers of your package write their code in</th>
+      <th>Your options</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td rowSpan="2">CJS source code using <code>require()</code></td>
+      <td>ESM: consumers <code>import</code> your package</td>
+      <td><a href="#cjs-source-and-only-esm-distribution">CJS source and only ESM distribution</a></td>
+    </tr>
+    <tr>
+      <td>CJS & ESM: consumers either <code>require()</code> or <code>import</code> your package</td>
+      <td><a href="#cjs-source-and-both-cjs-amp-esm-distribution">CJS source and both CJS & ESM distribution</a></td>
+    </tr>
+    <tr>
+      <td rowSpan="2">ESM source code using <code>import</code></td>
+      <td>CJS: consumers <code>require()</code> your package (and you use top-level <code>await</code>)</td>
+      <td><a href="#esm-source-with-only-cjs-distribution">ESM source with only CJS distribution</a></td>
+    </tr>
+    <tr>
+      <td>CJS & ESM: consumers either <code>require()</code> or <code>import</code> your package</td>
+      <td><a href="#esm-source-and-both-cjs-amp-esm-distribution">ESM source and both CJS & ESM distribution</a></td>
+    </tr>
+  </tbody>
+</table>
+
+### General notes
+
+[Syntax detection](https://nodejs.org/api/packages.html#syntax-detection) is _**not**_ a replacement for proper package configuration; syntax detection is not fool-proof and it has [significant performance cost](https://github.com/nodejs/node/pull/55238).
+
+When using [`"exports"`](https://nodejs.org/api/packages.html#conditional-exports) in package.json, it is generally a good idea to include `"./package.json": "./package.json"` so that it can be imported ([`module.findPackageJSON`](https://nodejs.org/api/module.html#modulefindpackagejsonspecifier-base) is not affected by this limitation, but `import` may be more convenient).
+
+`"exports"` can be advisable over [`"main"`](https://nodejs.org/api/packages.html#main) because it prevents external access to internal code (so you can be relatively sure users are not depending on things they shouldn't). If you don't need that, `"main"` is simpler and may be a better option for you.
+
+The `"engines"` field provides both a human-friendly and a machine-friendly indication of which version(s) of Node.js the package is compatible. Depending on the package manager used, an exception may be thrown causing the installation to fail when the consumer is using an incompatible version of Node.js (which can be very helpful to consumers). Including this field will save a lot of headache for consumers with an older version of Node.js who cannot use the package.
+
+### CJS source and distribution
+
+You _technially_ may not need _any_ package configuration beyond [`"name"`](https://nodejs.org/api/packages.html#name). But the less arcane, the better: Essentially just declare the package’s exports via the `"exports"` field/field-set.
```

Comment:
```suggestion
You _technially_ may not need _any_ package configuration beyond [`"name"`](https://nodejs.org/api/packages.html#name). But the less arcane, the better: Essentially just declare the package’s exports via the `"exports"` field/field-set.
```

I'd rather say.
The most minimal configuration is: (just names) but it's better to explictly declare exports because....